python/pyspark/sql/dataframe.py

```diff
@@ -437,10 +437,11 @@ def hint(self, name, *parameters):
         if not isinstance(name, str):
             raise TypeError("name should be provided as str, got {0}".format(type(name)))
 
+        allowed = [str, list, float, int]
         for p in parameters:
-            if not isinstance(p, str):
+            if not type(p) in allowed:
```

Comment:
I think isinstance is preferred, string can be str or unicode etc..

Reply:
Oh good to know, good catch. But then should I replicate `isinstance` for the other types (int, float, etc)? Or adding unicode to `allowed` is also a solution?

Reply:
Hm? can't we just simply do `isinstance(p, (basestring, list, float, int))`?

Reply:
Didn't know that it was possible, nice!